<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from urdf/hopper_1d.xacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="hopper" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">
  <link name="base_link"/>
  <joint name="joint_z" type="prismatic">
    <parent link="base_link"/>
    <child link="hip"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="hip">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.07 0.07 0.07"/>
      </geometry>
      <material name="light_grey">
        <color rgba=".7 .7 .7 .8"/>
      </material>
    </visual>
    <inertial>
      <mass value="0.15"/>
      <inertia ixx="6.125e-05" ixy="0.0" ixz="0.0" iyy="6.125e-05" iyz="0.0" izz="6.125e-05"/>
    </inertial>
  </link>

  <joint name="joint_hip" type="revolute">
    <parent link="hip"/>
    <child link="shank"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.035"/>
  </joint>
  <link name="shank">
     <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>

      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="joint_knee" type="revolute">
    <parent link="shank"/>
    <child link="lower_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="lower_leg">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="joint_contact" type="fixed">
    <parent link="lower_leg"/>
    <child link="contact"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="contact">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="red">
        <color rgba="1. 0. 0. 1."/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0." ixy="0.0" ixz="0.0" iyy="0." iyz="0.0" izz="0."/>
    </inertial>
  </link>
</robot>
